APMrover2/AP_Arming.cpp

```diff
@@ -185,3 +186,13 @@ bool AP_Arming_Rover::parameter_checks(bool report)
     return true;
 }
 
+// check if arming allowed from this mode
+bool AP_Arming_Rover::mode_checks(bool report)
+{   
+    //display failure if arming in this mode is not allowed
+    if ((rover.control_mode->mode_number()== Mode::Number::RTL) || (rover.control_mode->mode_number()== Mode::Number::SMART_RTL)) {
```

Comment:
Please create an `virtual bool allows_arming() const` method on `Mode`.  It should also be true for `INITIALIZING`, too.  The pattern is already present - you used a similar method last iteration :-)

We try to avoid sprinkling specific modes through the code as these lists tend to get forgotten when adding new modes.

Also, check the pointers rather than the mode numbers - so `rover.control_mode == &rover.mode_rtl` (not that the refactoring will make this necessary).